In [1]:
import numpy as np
import tensorflow as tf

In [19]:
coefficient=np.array([[1.],[-10.],[25.]]) #the data will plug into x

w= tf.Variable(0,dtype=tf.float32) 
x=tf.placeholder(tf.float32, [3,1]) #training set
#cost= tf.add(tf.add(w**2,tf.multiply(-10.,w)),25) #W^2-10W+25W
#cost=w**2 - 10*w +25
cost=x[0][0]*w**2 + x[1][0]*w +x[2][0] #cost function is important
cost

<tf.Tensor 'add_2:0' shape=() dtype=float32>

In [20]:
train=tf.train.GradientDescentOptimizer(0.01).minimize(cost) #0.01 is learning rate
train

<tf.Operation 'GradientDescent_3' type=NoOp>

In [21]:
init=tf.global_variables_initializer()
session = tf.Session()
session.run(init)
print(session.run(w)) #for TensorFlow's evaluative variable

0.0


w equals to 0 because we haven't run anything yet

In [22]:
session.run(train, feed_dict={x:coefficient}) #run one step of gradient descent
print(session.run(w)) #evaluate the vale of w after one step of gradient descent

0.1


let's run 1000 times

In [24]:
for i in range(1000):
    session.run(train, feed_dict={x:coefficient})
    
print(session.run(w))

4.99999


because we minimize the $(m-5)^2$ ,so 4.99999 is very close